pkg/profile/helm.go

```diff
@@ -0,0 +1,111 @@
+package profile
+
+import (
+	"reflect"
+	"strings"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// helmReleaseRequiresUpdate checks if the helm release resource requires updating
+func helmReleaseRequiresUpdate(existingRes, newRes *helmv2.HelmRelease) bool {
+	switch {
+	case existingRes.Spec.Chart.Spec.Chart != newRes.Spec.Chart.Spec.Chart:
+		return true
+	case existingRes.Spec.Chart.Spec.Version != newRes.Spec.Chart.Spec.Version:
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Chart.Spec.SourceRef, newRes.Spec.Chart.Spec.SourceRef):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Values, newRes.Spec.Values):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.ValuesFrom, newRes.Spec.ValuesFrom):
+		return true
+	default:
+		return false
+	}
+}
+
+// HelmRepoRequiresUpdate checks if the helm repository resource requires updating
+func helmRepoRequiresUpdate(existingRes, newRes *sourcev1.HelmRepository) bool {
+	switch {
+	case existingRes.Spec.URL != newRes.Spec.URL:
+		return true
+	default:
+		return false
+	}
+}
+
+func (p *Profile) makeHelmRepository(url string, name string) *sourcev1.HelmRepository {
+	return &sourcev1.HelmRepository{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.makeHelmRepoName(name),
+			Namespace: p.subscription.ObjectMeta.Namespace,
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       sourcev1.HelmRepositoryKind,
+			APIVersion: sourcev1.GroupVersion.String(),
+		},
+		Spec: sourcev1.HelmRepositorySpec{
+			URL: url,
+		},
+	}
+}
+
+func (p *Profile) makeHelmRepoName(name string) string {
+	repoParts := strings.Split(p.subscription.Spec.ProfileURL, "/")
+	repoName := repoParts[len(repoParts)-1]
+	return join(p.subscription.Name, repoName, p.subscription.Spec.Branch, name)
+}
```

Comment:
This is identical to `makeGitRepoName`. This can be renamed to `makeRepoName` and reused for both Git and Helm repos.

Reply:
I just moved it as part of this PR, but yeah thats a fair refactor I'll add in

Reply:
The difference is that it adds the `name` at the join. And it looks awkward to call `makeRepoName("")`. But it could be two functions but a single function being called inside those two functions.

Reply:
Actually that wouldn't work. It would end up being `asdf-asdf-` with an extra `-` at the end.

Reply:
If you use this function for two elements they will have the same name which will cause a name collision.

Reply:
Oh! But you need it. Because if you have a helmchart which also generates the same resource, and you don't add the name which is the Chart's name then the name will be the same for the same resource.